client/src/modules/stock/StockForm.service.js

```diff
@@ -52,12 +52,11 @@ function StockFormService(StockItem, Store, AppCache, Session, $timeout) {
    */
   StockForm.prototype.addItems = function addItems(n) {
     var elt;
-    var idx;
-    var self = this;
+
+    // will repeat will n > 0
     while (n--) {
       elt = new StockItem();
-      idx = self.store.data.length;
-      elt.index = idx ? idx + 1 : idx;
+      elt.index = this.store.data.length;
```

Comment:
You are right, this code was already there.
I think index is the name chosen to define a key of the object to store and the value does not matter because it is just an integer.

Reply:
I decided to use ```id```